modules/vertice-cor-report/main.tf

```diff
@@ -0,0 +1,45 @@
+locals {
+  defaults_columns_for_selection = ["account_id", "action_type", "currency_code", "current_resource_details", "current_resource_summary",
+    "current_resource_type", "estimated_monthly_cost_after_discount", "estimated_monthly_cost_before_discount",
+    "estimated_monthly_savings_after_discount", "estimated_monthly_savings_before_discount",
+    "estimated_savings_percentage_after_discount", "estimated_savings_percentage_before_discount",
+    "implementation_effort", "last_refresh_timestamp", "recommendation_id", "recommendation_lookback_period_in_days",
+    "recommendation_source", "recommended_resource_details", "recommended_resource_summary", "recommended_resource_type",
+  "region", "resource_arn", "restart_needed", "rollback_possible", "tags"]
+  cor_columns_for_selection = length(var.cor_columns_for_selection) == 0 ? join(", ", local.defaults_columns_for_selection) : join(", ", setintersection(local.defaults_columns_for_selection, var.cor_columns_for_selection))
```

Comment:
With the `setintersection` here, some column (e.g., if there is a typo in some of the variable values passed) might silently be dropped without the user realising.

I think it might be better to validate that all the values are valid...